libtransact/src/workload/source.rs

```diff
@@ -16,48 +16,111 @@
  * ------------------------------------------------------------------------------
  */
 
-//! Tools for generating signed batches from a stream of transactions
+//! Tools for reading length-delimited protobuf from a file
 
 use std::io::Read;
-use std::marker::PhantomData;
 
 use protobuf::Message;
 
-/// Decodes Protocol Buffer messages from a length-delimited input reader.
-pub struct LengthDelimitedMessageSource<'a, T: 'a> {
+use crate::error::InvalidStateError;
+use crate::protocol::batch::BatchPair;
+use crate::protocol::transaction::Transaction;
+use crate::protos::FromProto;
+use crate::workload::{BatchReader, TransactionReader};
+
+/// Decodes Protocol Buffer `Batches` from a length-delimited input reader.
+pub struct ProtobufBatchReader<'a> {
```

Comment:
Put ProtobufBatchReader/ProtobufBatchWriter in a separate module (or two) so it cleans up the imports (it should be clear that only that code pulls in anything from protos.

Reply:
The traits are now in their own `batch_reader` and `transaction_reader` modules each with their own `protobuf` sub-module containing the protobuf impls